ls1intum/Artemis
# Assessment of team submissions by team owner tutor

### Checklist

### Motivation and Context
Tutors need to be able to assess student submissions in team-based exercises. The concept here is  different than for regular exercises where tutors receive random submissions to assess without knowing to which students they belong. For team-based exercises, each team has a tutor assigned (also referred to as the team owner in technical terms) who acts as the team mentor and will assess the submissions of his or her teams (likely across all team-based exercises in a course). 

### Description
Since this assessment flow is substantially different than the regular one, we decided on a different solution here for now than using the Tutor Exercise Dashboard. It works as follows: On the Tutor Course Dashboard, team-based exercises have a button "Exercise Teams" instead of the "Exercise Dashboard" button for regular individual exercises. When clicked, the tutor is presented with all teams assigned to him for that exercise. He or she can now click onto the different teams and see their participations in all exercises for the course. The rightmost column "Assessment" contains a button for the tutor to assess the students' submission.

### Steps for Testing


1. Log in to Artemis as a tutor
2. Navigate to Course Management
3. Click on Tutor Course Dashboard for a Course with Team Exercises
4. Click on the button "Exercise Teams" for one of the Team Exercises (in which you are the team owner of several teams)
5. Open several of your teams by clicking on the short names of the teams
6. Check that the new assessment column is present in the participations table for the team
7. Check that the label of the assessment button is correct depending on the state of the submission and result (button should be disabled if there is no submission, otherwise it should be enabled and say "Start new assessment" initially, then "Continue assessment" once a result exists and then finally "Open assessment" if the result has a completion date)

### Screenshots
If a tutor clicks on "Exercise Teams" on the Tutor Course Dashboard, he will get to an overview of his assigned teams for the exercise.
![OnPaste 20200518-144221](https://user-images.githubusercontent.com/7109225/82214231-cebbd280-9915-11ea-906d-860568f1c588.png)

This uses the regular teams overview page but has the new filter "Own teams" enabled automatically when coming via the Tutor Course Dashboard.
![OnPaste 20200518-144144](https://user-images.githubusercontent.com/7109225/82214234-cf546900-9915-11ea-8e6f-7abd1c990ce5.png)

Having clicked on a team short name, the team page is shown with the new assessment column to the right:
![OnPaste 20200517-201519](https://user-images.githubusercontent.com/7109225/82156675-eb57fc00-987c-11ea-9102-0a172ecb131c.png)
